practicas/David/p1/1_niveles_linguisticos.py

```diff
@@ -0,0 +1,1214 @@
+#         - Total de palabras
+#         - La longitud de palabra promedio
+#         - La cuenta de morfemas promedio
+#         - La categoría más común
+#     - Con base en esta información elabore una conclusión lingüística sobre la morfología de las lenguas analizadas.
+# -
+
+data_langs = {LANGS[lang]: raw_corpus_to_dataframe(get_raw_corpus(get_track_files(lang)), lang) for lang in LANGS.keys()}
+
+data_langs["Spanish"].head()
+
+# +
+fig = plt.figure(layout='constrained', figsize=(30, 20))
+fig.suptitle('Análisis estadístico descriptivo de los datos morfológicos', fontsize=20)
+
+subfigs = fig.subfigures(2, 4)
+
+colors = plt.cm.tab10.colors
+
+for i, (lang, data) in enumerate(data_langs.items()):
+    
+    subfigs[i // 4, i % 4].set_facecolor('lightgrey')
+    if len(data['category'].unique()) == 1:
+        ax = subfigs[i // 4, i % 4].subplots(2,1)
+    else:
+        ax = subfigs[i // 4, i % 4].subplots(3,1)
+        
+    ax[0].hist(data['word_len'], bins=int(len(data['word_len'].unique())*0.95), edgecolor='black', color=colors[i])
+    ax[0].set_title(f'Word Length Distribution for {lang}')
+    ax[0].set_xlabel('Word Length')
+    ax[0].set_ylabel('Frequency')
+    
+    if 1 < len(data['category'].unique()):
+        ax[1].bar(data['category'].value_counts().index, data['category'].value_counts().values, color=colors[i])
+        ax[1].set_title(f'Category Frequency for {lang}')
+        ax[1].set_xlabel('Category')
+        ax[1].set_ylabel('Frequency')
+        ax[1].tick_params(axis='x', rotation=45)
+        
+        ax[2].bar(data['morph_count'].value_counts().index, data['morph_count'].value_counts().values, color=colors[i])
+        ax[2].set_title(f'Morpheme Count Distribution for {lang}')
+        ax[2].set_xlabel('Morpheme Count')
+        ax[2].set_ylabel('Frequency')
+    
+    else:
+        ax[1].bar(data['morph_count'].value_counts().index, data['morph_count'].value_counts().values, color=colors[i])
+        ax[1].set_title(f'Morpheme Count Distribution for {lang}')
+        ax[1].set_xlabel('Morpheme Count')
+        ax[1].set_ylabel('Frequency')
+plt.show()
+
+
+# -
+
+def statistics_lang(lang: str) -> list[int, float, float, str]:
+    """
+    Computes summary statistics for a given language from the `data_langs` dataset.
+
+    Parameters:
+    lang (str): The language key to retrieve data from `data_langs`.
+
+    Returns:
+    list[int, float, float, str]: A list containing:
+        - The total count of words (int).
+        - The average word length (float).
+        - The average number of morphemes per word (float).
+        - The most common category (str).
+    """
+    data = data_langs[lang]
+    summarized = data.agg({'words':'count', 'word_len':'mean', 'morph_count':'mean'})
+    mode = data_langs[lang]['category'].mode()[0]
+    return [int(summarized['words']), summarized['word_len'], summarized['morph_count'], mode]
+
+
+for lang in LANGS.values():
+    print([lang] + statistics_lang(lang))
```

Comment:
Estaría bueno indicar a que corresponde cada elemento de la lista
```suggestion
   print("lang\twords\tword_len_mean\tmorph_count_mean\tmode_cat")
for lang in LANGS.values():
    print([lang] + statistics_lang(lang))
```